lib/ansible/modules/cloud/vmware/vmware_guest_disk.py

```diff
@@ -0,0 +1,604 @@
+                pass
+        datastore = None
+        datastore_freespace = 0
+        for ds in datastore_cluster_obj.childEntity:
+            if isinstance(ds, vim.Datastore) and ds.summary.freeSpace > datastore_freespace:
+                # If datastore field is provided, filter destination datastores
+                datastore = ds
+                datastore_freespace = ds.summary.freeSpace
+        if datastore:
+            return datastore.name
+        return None
+
+    @staticmethod
+    def gather_disk_facts(vm_obj):
+        """
+        Function to gather facts about VM's disks
+        Args:
+            vm_obj: Managed object of virtual machine
+
+        Returns: A list of dict containing disks information
+
+        """
+        disks_facts = dict()
+        if vm_obj is None:
+            return disks_facts
+
+        disk_index = 0
+        for disk in vm_obj.config.hardware.device:
+            if isinstance(disk, vim.vm.device.VirtualDisk):
+                disks_facts[disk_index] = dict(
+                    key=disk.key,
+                    label=disk.deviceInfo.label,
+                    summary=disk.deviceInfo.summary,
+                    backing_filename=disk.backing.fileName,
+                    backing_datastore=disk.backing.datastore.name,
+                    backing_disk_mode=disk.backing.diskMode,
+                    backing_writethrough=disk.backing.writeThrough,
+                    backing_thinprovisioned=disk.backing.thinProvisioned,
+                    backing_eagerlyscrub=bool(disk.backing.eagerlyScrub),
+                    controller_key=disk.controllerKey,
+                    unit_number=disk.unitNumber,
+                    capacity_in_kb=disk.capacityInKB,
+                    capacity_in_bytes=disk.capacityInBytes,
+                )
+                disk_index += 1
+        return disks_facts
+
+
+def main():
+    argument_spec = vmware_argument_spec()
+    argument_spec.update(
+        name=dict(type='str'),
+        uuid=dict(type='str'),
+        folder=dict(type='str'),
+        datacenter=dict(type='str', required=True),
+        disk=dict(type=list, default=[]),
+    )
+    module = AnsibleModule(argument_spec=argument_spec,
+                           required_one_of=[['name', 'uuid']])
+
+    if module.params['folder']:
+        # FindByInventoryPath() does not require an absolute path
+        # so we should leave the input folder path unmodified
+        module.params['folder'] = module.params['folder'].rstrip('/')
+
+    pyv = PyVmomiHelper(module)
+    # Check if the VM exists before continuing
+    vm = pyv.get_vm()
+
+    if vm:
+        # VM exists
+        pyv.ensure_disks(vm_obj=vm)
+        # try:
+        #    pyv.ensure_disks(vm_obj=vm)
+        # except Exception as exc:
+        #    module.fail_json(msg="Failed to manage disks for virtual machine"
+        #                        " '%s' with exception : %s" % (vm.name, to_native(exc)))
```

Comment:
Those comments don't bring any information, maybe you could remove then. What do you think ?

Reply:
OK. Leftover debugging stuff